x-pack/plugins/observability_ai_assistant/public/hooks/use_json_editor_model.ts

```diff
@@ -25,15 +26,22 @@ export const useJsonEditorModel = ({
 
   const functionDefinition = chatService.getFunctions().find((func) => func.name === functionName);
 
+  const [initialJsonValue, setInitialJsonValue] = useState<string | undefined>(initialJson);
+
+  useEffect(() => {
+    setInitialJsonValue(initialJson);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [functionName]);
```

Comment:
I wonder if you can simplify this by using `const initialJson = useOnce(initialJsonFromProps)`. Another question here:  do we actually want to set the `initialJsonValue` to `initialJson` if the function name changes? Shouldn't we wipe it in that case?

Reply:
> Another question here: do we actually want to set the initialJsonValue to initialJson if the function name changes? Shouldn't we wipe it in that case?

We could do that, in this case the payload that the user entered before will be displayed, but the editor will show to the user the payload is invalid as it most likely will not comply with the schema of the new function that the user selected. This would allow the user to keep his payload and tweak bits of it, but this might also be overkill.

Reply:
Just double checked and the payload is cleared and prefilled with the schema defaults when the user selects a new function while editing an existing function.